src/Helpers/LanguageProvider.php

```diff
@@ -4,6 +4,24 @@
 
 class LanguageProvider
 {
+    /**
+     * Register the current subdomain, so that Polylang
+     * can recognize it and localize the api output as expected.
+     */
+    public static function registerSubdomain()
+    {
+        if (!is_admin()) {
+            $options = get_option('polylang');
+            foreach ($options['domains'] as $locale => $domain) {
+                if (str_contains($_SERVER['HTTP_HOST'], parse_url($domain, PHP_URL_HOST))) {
+                    $subDomain = sprintf('%s://%s', parse_url($domain, PHP_URL_SCHEME), $_SERVER['HTTP_HOST']);
+                    $options['domains'][$locale] = $subDomain;
+                }
+            }
+            update_option('polylang', $options);
```

Comment:
I don't think we can use update_option() as this will overwrite the domains set in the settings page